File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/domain/CartCatalogUseCaseTest.java
package com.tarbadev.witchcraft.domain;

import com.tarbadev.witchcraft.domain.entity.Cart;
import com.tarbadev.witchcraft.domain.repository.CartRepository;
import com.tarbadev.witchcraft.domain.usecase.CartCatalogUseCase;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;
import org.springframework.test.context.ActiveProfiles;

import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Java6Assertions.assertThat;
import static org.mockito.BDDMockito.given;

@RunWith(MockitoJUnitRunner.class)
@ActiveProfiles("test")
public class CartCatalogUseCaseTest {
  @Mock private CartRepository cartRepository;

  private CartCatalogUseCase subject;

  @Before
  public void setUp() {
    subject = new CartCatalogUseCase(cartRepository);
  }

  @Test
  public void execute() {
    List<Cart> carts = Arrays.asList(
        Cart.builder().build(),
        Cart.builder().build(),
        Cart.builder().build(),
        Cart.builder().build()
    );

    given(cartRepository.findAll()).willReturn(carts);

    assertThat(subject.execute()).isEqualTo(carts);
  }
}

File: WitchcraftApp/src/test/kotlin/com/tarbadev/witchcraft/carts/domain/usecase/CartCatalogUseCaseTest.kt
package com.tarbadev.witchcraft.carts.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.reset
import com.nhaarman.mockitokotlin2.whenever
import com.tarbadev.witchcraft.carts.domain.entity.Cart
import com.tarbadev.witchcraft.carts.domain.repository.CartRepository
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

import java.util.Arrays

class CartCatalogUseCaseTest {
    private val cartRepository: CartRepository = mock()
    private var cartCatalogUseCase: CartCatalogUseCase = CartCatalogUseCase(cartRepository)

    @BeforeEach
    fun setUp() {
        reset(cartRepository)
    }

    @Test
    fun execute() {
        val carts = Arrays.asList(
                Cart(),
                Cart(),
                Cart(),
                Cart()
        )

        whenever(cartRepository.findAll()).thenReturn(carts)

        assertEquals(carts, cartCatalogUseCase.execute())
    }
}